### PR TITLE
feat(nx-plugin,vite-utils): Use relevant polyfills instead of all

### PR DESCRIPTION
**Pull request by luxaritas**
We're currently using the "all" polyfill-io featureset and the following is logged to the console:
> Using the `all` alias with polyfill.io is a very bad idea. In a future version of the service, `all` will deliver the same behaviour as `default`, so we recommend using `default` instead.

I've implemented a Vite plugin that automatically injects the correct polyfill-io script based on actual used js features in the final output bundle as determined by https://github.com/Financial-Times/js-features-analyser/ (based on https://github.com/Financial-Times/polyfill-service-url-builder).

I've also removed the path aliases from the vite-utils tsconfig as this isn't functional (because the vite config for this package uses this package, when the config gets initially bundled to build itself, it doesn't have aliases available)

### CODE REVIEW

**Comment by luxaritas** [comment]
In retrospect, https://github.com/Financial-Times/polyfill-service-url-builder would probably be more bulletproof and be a lower maintenance burden. I'd need to investigate how to integrate with Vite - might need to actually call js-features-analyser directly under the hood with the raw bundle(s). If we don't have time to do it before this gets merged, should file an issue about it

**Comment by luxaritas** [comment]
I've now implemented the plugin to autogenerate the polyfill-io URL. I'll update the PR description to reflect the current state here

File: packages/vite-utils/tsconfig.build.json [hunk 1]
```diff
@@ -1,10 +1,5 @@
 {
   "extends": "@eternagame/tsconfig/tsconfig.node.json",
-  "compilerOptions": {
```
Comment (tkaragianes) [inline packages/vite-utils/tsconfig.build.json:3]: We get rid of path aliasing because the vite-utils package never uses it?

**Review by tkaragianes — APPROVED** [review]

File: packages/vite-utils/tsconfig.build.json (hunk 1 above)
Comment (luxaritas) [reply packages/vite-utils/tsconfig.build.json:3]: Not just never uses it - can't use it. When we run `vite build` for vite-utils, vite internally compiles the vite.config.ts. Vite/esbuild doesn't support tsconfig path aliases - that's something we configure in the vite config itself. But on first build, the compilation of vite.config.ts to build vite-utils (while the config itself uses vite-utils!) will not be using our vite config, and so will not have any aliases registered.

**Pull request merged**